txnsync/peerscheduler.go

```diff
@@ -97,6 +97,21 @@ func (p *peerScheduler) nextPeers() (outPeers []*Peer) {
 		bucket := heap.Remove(p, 0).(peerBucket)
 		outPeers = append(outPeers, bucket.peer)
 	}
+
+	// in many cases, we'll have only a single peer; however, in case we have multiple
+	// ( which is more likely when we're "running late" ), we want to make sure to remove
+	// duplicate ones.
```

Comment:
Are you doing this here to avoid making a map each time `Push` is called? It's a bit confusing to do it this way, and don't you also need to remove duplicate peers from the `p.peers`?

Reply:
That's a good question. The reason for this is that the `next` timestamp ( which is now the `p.peers[0].next` ) is going to be dynamically throttled to allow different "groups" of peers to be selected. Because of that, it's not possible to do that during the `Push`.

This PR is one of a series of PR required for implementing the throttler. So - it lays the foundation for the throttler without introducing the actual implementation.

Reply:
and no, there is no need to remove duplicate `p.peers` instance, since we want to have the ability to have multiple-schedules for a single peer.